SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,292 @@
+      DataT *ref_data_for_change_ptr = ref_data_for_change.data();
+      DataT *const out_ptr = result.data();
+      auto is_correct_type_ptr = is_correct_type.data();
+
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, NumSelectedElemsT, StrideT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            *is_correct_type_ptr =
+                TestCaseT::template call_operator<DataT, NumElems,
+                                                  NumSelectedElems, Stride>(
+                    init_ref_ptr, ref_data_for_change_ptr, out_ptr, Offset);
+          });
+    });
+    queue.wait_and_throw();
+
+    std::vector<size_t> selected_indexes;
+    // Collect the indexess that has been selected.
+    for (size_t i = Offset; i < Stride * NumSelectedElems + Offset;
+         i += Stride) {
+      selected_indexes.push_back(i);
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto next_selected_index = selected_indexes.begin();
+
+    // Verify that values, that do not was selected has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been selected and changed.
+      if (i < *next_selected_index) {
+        const DataT &expected = initial_ref_data[i];
+        const DataT &retrieved = result[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+      } else {
+        const DataT &expected = ref_data_for_change[(i - Offset) / Stride];
+        const DataT &retrieved = result[i];
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type);
+        }
+        next_selected_index++;
+      }
+    }
+
+    if (!is_correct_type.value()) {
+      passed = false;
+      log::fail(TestDescriptionT(data_type), "Unexpected return type.");
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type) {
+    log::fail(TestDescriptionT(data_type), "Unexpected value at index ", i,
+              ", retrieved: ", retrieved, ", expected: ", expected);
+
+    return false;
+  }
+};
+
+// Aliases to provide size or stride values to test.
+// This is the syntax sugar just for code readability.
+template <int N> using stride_type = std::integral_constant<int, N>;
+template <int N> using size_type = std::integral_constant<int, N>;
+template <int N> using offset_type = std::integral_constant<int, N>;
+
+template <typename SelectT, int NumSelectedElems, int Stride, int Offset,
+          typename... T, int... Values>
+bool run_with_size_stride_offset(
+    sycl::queue &queue, const named_type_pack<T...> &types,
+    const unnamed_type_pack<std::integral_constant<int, Values>...> &sizes) {
```

Comment:
Why not just `std::forward` parameters passed? We are not using `queue`, `types`, `sizes` or `Values` here anyway.
```suggestion
bool run_with_size_stride_offset(ArgsT &&... args) {
```

Reply:
this comment applied [here](https://github.com/intel/llvm-test-suite/pull/875/commits/6e3c354f9cdd7463f3b5749cc615b9ff59eaa9ca#diff-295141c99d991cbb4971abc3bd754b2f17400cade3b579ac7bc36b28a1450dc3R222-R223)